src/encoded/static/components/matrix_encore.js

```diff
@@ -0,0 +1,619 @@
+            </div>
+        </div>
+    </div>
+);
+
+TargetFilter.propTypes = {
+    /** Current value of filtering text */
+    filterText: PropTypes.string,
+    /** Called when the user changes a search input field */
+    textChangeHandler: PropTypes.func.isRequired,
+};
+
+TargetFilter.defaultProps = {
+    filterText: '',
+};
+
+
+/**
+ * Render the area above the facets and matrix content.
+ */
+const MatrixHeader = ({ context, targetFilterText, textChangeHandler }) => (
+    <div className="matrix-header">
+        <div className="matrix-header__title">
+            <div className="matrix-title-badge">
+                <h1>{context.title}</h1>
+                <MatrixBadges context={context} />
+            </div>
+            <div className="matrix-description">
+                <div className="matrix-description__text">
+                    The ENCORE project aims to study protein-RNA interactions by creating a map of RNA binding proteins (RBPs) encoded in the human genome and identifying the RNA elements that the RBPs bind to.
+                </div>
+            </div>
+        </div>
+        <div className="matrix-header__controls">
+            <div className="matrix-header__target-filter-controls">
+                <TargetFilter filterText={targetFilterText} textChangeHandler={textChangeHandler} />
+            </div>
+        </div>
+    </div>
+);
+
+MatrixHeader.propTypes = {
+    /** Matrix search result object */
+    context: PropTypes.object.isRequired,
+    /** Current value of filtering text */
+    targetFilterText: PropTypes.string,
+    /** Called when the user changes a search input field */
+    textChangeHandler: PropTypes.func.isRequired,
+};
+
+MatrixHeader.defaultProps = {
+    targetFilterText: '',
+};
+
+
+/**
+ * Custom react hook to handle matrix scrolling.
+ * @param {node} Ref to scrolling matrix element
+ *
+ * @return {array} States and actions usable to this hook's clients
+ */
+const useMatrixScrollHandler = (scrollElement) => {
+    const [scrolledRight, setScrolledRight] = React.useState(true);
+
+    /**
+     * Show a scroll indicator depending on current scrolled position.
+     * @param {object} element DOM element to apply shading to
+     */
+    const handleScrollIndicator = React.useCallback((element) => {
+        // Have to use a "roughly equal to" test because of an MS Edge bug mentioned here:
+        // https://stackoverflow.com/questions/30900154/workaround-for-issue-with-ie-scrollwidth
+        const scrollDiff = Math.abs((element.scrollWidth - element.scrollLeft) - element.clientWidth);
+        if (scrollDiff < 2 && !scrolledRight) {
+            // Right edge of matrix scrolled into view.
+            setScrolledRight(true);
+        } else if (scrollDiff >= 2 && scrolledRight) {
+            // Right edge of matrix scrolled out of view.
+            setScrolledRight(false);
+        }
+    }, [scrolledRight]);
+
+    React.useEffect(() => {
+        if (scrollElement.current) {
```

Comment:
should this be:
if (scrollElement && scrollElement.current)
?

Reply:
scrollElement gets assigned from useRef on every render, so it has to exist as an object.